rllib/models/tf/tf_action_dist.py

```diff
@@ -64,32 +64,25 @@ def deterministic_sample(self) -> TensorType:
 
     @override(ActionDistribution)
     def logp(self, x: TensorType) -> TensorType:
-        return -tf.nn.sparse_softmax_cross_entropy_with_logits(
-            logits=self.inputs, labels=tf.cast(x, tf.int32)
-        )
+        logp = tf.math.log(tf.nn.softmax(self.inputs))
+        return tf.gather(logp, x, batch_dims=1)
 
     @override(ActionDistribution)
     def entropy(self) -> TensorType:
-        a0 = self.inputs - tf.reduce_max(self.inputs, axis=1, keepdims=True)
```

Comment:
The weird implementations you are referring to are actually modified versions of a softmax.
I've not done the original implementation, but I think that this line is supposed to add some numerical stability to the softmax. But from where I stand it looks more like it would add instabilities.

Consider _self.inputs = [n, m]_ where n and m are large negative numbers. ea0 would be close to zero and z0 would also be close to zero, with p0 being instable.

@kouroshHakha do you have some insight here?

Reply:
That being said, purely mathematically speaking, the softmax is invariant to translation.

Reply:
I've produced [a script](https://gist.github.com/ArturNiederfahrenhorst/b721c0085457c4db845542519d8f8f84) that shows differences in results.

Reply:
I approve this this change to entropy()

Reply:
Awesome. Thanks for the comparison script @ArturNiederfahrenhorst !
I have no objections to merging, other than my comment below on the sampling op. I think we need to change it to make action masking still work.